drv/opentitan-aontimer/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+            clock_freq_hz,
+            bark_cb,
+        };
+        unsafe {
+            // This is just a counter, reset it to 0.
+            (*inst.base_addr)[WDOG_COUNT_IDX] = 0;
+            // write 1 to clear any pending interrupts
+            (*inst.base_addr)[INTR_STATE_IDX] =
+                INTR_STATE_BITS_WKUP | INTR_STATE_BITS_WDOG;
+            (*inst.base_addr)[WDOG_BARK_THOLD_IDX] =
+                inst.ms_to_reg_count(bark_thold_ms)?;
+            if bark_thold_ms > bite_thold_ms {
+                return Err(());
+            }
+            (*inst.base_addr)[WDOG_BITE_THOLD_IDX] =
+                inst.ms_to_reg_count(bite_thold_ms)?;
+        }
+        Ok(inst)
+    }
+
+    //Disable the watchdog timer. If the configuration is locked, returns false.
+    pub fn disable(&self) -> Result<(), ()> {
+        if !self.check_lock() {
+            return Err(());
+        }
+        unsafe {
+            // disable everything
+            (*self.base_addr)[WDOG_CTRL_IDX] = 0;
+        }
+        Ok(())
+    }
+
+    //Enables the watchdog timer. If the configuration is locked, returns false.
+    pub fn enable(&self) -> Result<(), ()> {
+        if !self.check_lock() {
+            return Err(());
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_CTRL_IDX] |= WDOG_CTRL_BITS_EN;
+        }
+        Ok(())
+    }
+
+    //Enables the watchdog timer, then locks the configuration. If the
+    // configuration is already locked, returns false.
+    pub fn enable_and_lock(&self) -> Result<(), ()> {
+        if !self.check_lock() {
+            return Err(());
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_CTRL_IDX] |= WDOG_CTRL_BITS_EN;
+            (*self.base_addr)[WDOG_REGWEN_IDX] = 0;
+        }
+        Ok(())
+    }
+
+    //Sets the bark threshold in ms. If the configuration is locked, returns false.
+    pub fn set_bark_thold(&self, bark_thold_ms: u64) -> Result<(), ()> {
+        if !self.check_lock() {
+            return Err(());
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_BARK_THOLD_IDX] =
+                self.ms_to_reg_count(bark_thold_ms)?;
+        }
+        Ok(())
+    }
+
+    //Sets the bite threshold in ms. If the configuration is locked, returns false.
+    pub fn set_bite_thold(&self, bite_thold_ms: u64) -> Result<(), ()> {
+        if !self.check_lock() {
+            return Err(());
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_BITE_THOLD_IDX] =
+                self.ms_to_reg_count(bite_thold_ms)?;
+        }
+        Ok(())
+    }
+
+    //Pets the watchdog, i.e. resets the count down to 0.
+    pub fn feed_sacrifice(&self) {
+        unsafe {
+            (*self.base_addr)[WDOG_COUNT_IDX] = 0;
+        }
+    }
+
+    pub fn clear_wdt_irq(&self) {
+        unsafe {
+            (*self.base_addr)[INTR_STATE_IDX] =
+                INTR_STATE_BITS_WKUP | INTR_STATE_BITS_WDOG;
```

Comment:
Name suggestes that watchdog interrupt is being cleared, but the code actually clears both watchdog and wakeup interrupts. Why is this touching the wakeup timer bit?

Reply:
changed to just the watchdog.